loopchain/baseservice/peer_manager.py

```diff
@@ -110,80 +138,108 @@ def get_peer_ids_hash(peer_ids):
-        """
-        # util.logger.spam(f"peer_list({self.peer_list_data.peer_info_list})")
-        return self.peer_list_data.peer_info_list
+        request = convert_params(request, ParamType.call)
+        stub = StubCollection().icon_score_stubs[ChannelProperty().name]
+        response = cast(dict, stub.sync_task().call(request))
+        response_to_json_query(response)
 
-    @property
-    def peer_order_list(self) -> dict:
-        """
+        util.logger.debug(f"in load_peers_from_iiss response({response})")
+        if 'preps' not in response['result']:
+            util.logger.debug(f"There is no preps in result.")
+            return
 
-        :return: { order:peer_id }
-        """
-        return self.peer_list_data.peer_order_list
+        peer_ids = (preps["id"] for preps in response["result"]["preps"])
+        peer_ids_appended = reduce(add, peer_ids, '')
 
-    def set_peer_list(self, peer_list_data: PeerListData):
-        """ update PeerList
+        if self.get_peer_ids_hash(peer_ids_appended) == self.peer_ids_hash():
+            util.logger.debug(f"There is no change in peers.")
+            return
 
-        :param peer_list_data: PeerListData
-        """
-        self.peer_list_data = peer_list_data
+        if not conf.LOAD_PEERS_FROM_IISS:
+            return
+
+        util.logger.debug(f"Peer manager have to update with new list.")
+        self.reset_peers(check_status=False)
+
+        reps = response["result"]["preps"]
+        self._add_reps(reps)
+
+    async def load_peers_from_file(self):
+        channel_info = util.load_json_data(conf.CHANNEL_MANAGE_DATA_PATH)
+        reps: list = channel_info[ChannelProperty().name].get("peers")
+        for peer in reps:
+            self.add_peer(peer)
+
+    async def load_peers_from_rest_call(self):
+        response = ObjectManager().channel_service.radio_station_stub.call("GetReps")
+        reps = response.get('rep')
+        self._add_reps(reps)
+
+    def _add_reps(self, reps: list):
+        for order, rep_info in enumerate(reps, 1):
+            peer = Peer(rep_info["id"], rep_info["id"], rep_info["target"], order=order)
+            self.add_peer(peer)
+
+    def show_peers(self):
+        util.logger.debug(f"peer_service:show_peers ({ChannelProperty().name}): ")
+        for peer in self.get_IP_of_peers_in_group():
+            util.logger.debug("peer_target: " + peer)
+
+    def get_quorum(self):
+        peer_count = self.get_peer_count()
+        quorum = math.floor(peer_count * conf.VOTING_RATIO) + 1
+        complain_quorum = math.floor(peer_count * (1-conf.VOTING_RATIO)) + 1
+
+        return quorum, complain_quorum
+
+    def get_reps(self):
+        peer_ids = (self.peer_order_list[peer_order]
+                    for peer_order in sorted(self.peer_order_list.keys()))
+        peers = (self.peer_list[peer_id] for peer_id in peer_ids)
+        return [{"id": peer.peer_id, "target": peer.target} for peer in peers]
```

Comment:
Make lazy generators and turned to list when returns?

How about?:
```suggestion
        return ({"id": peer.peer_id, "target": peer.target} for peer in peers)
```

Reply:
The expected value of the caller of get_reps is list.